orderbook/openapi.yml

```diff
@@ -161,6 +161,24 @@ paths:
           description: Invalid signature
         404:
           description: Order was not found
+  /api/v1/orders/{txHash}:
```

Comment:
Instead of implementing a new route like this, I think from an API perspective it would make more sense to add a new query parameter to the existing `api/v1/orders` endpoint (https://github.com/gnosis/gp-v2-services/blob/7428f278d961a51a555bd4b940ddd7a1cf5e4804/orderbook/openapi.yml#L57-L109).

One reason I'm kind of reluctant to use this endpoint is that it feels not very "REST"-ty. Specifically, here, you aren't returning some transaction data (which is what the URL would suggest to me), but rather orders that match some transaction hash. As such I would suggest either:
- `api/v1/orders?txHash=0x...`: This is what I slightly prefer, although I understand that the maybe the existing `api/v1/orders` may be getting too complex. That being said, it would allow you to do things like "query all orders that sold token X in transaction Y": `api/v1/orders?sellToken=X&txHash=Y`.
- `api/v1/transactions/{txHash}/orders`: This way the URL would clearly indicate that orders are the data that is being returned. It would also add a more natural place to return trades that happened in a specific transaction: `api/v1/transactions/{txHash}/trades`

What do others think?

Also looping in @anxolin and @alfetopito

Reply:
100% agree with @nlordell , I would have written exatly the same (but I would've explained worse)

problem with `/api/v1/orders/{txHash}` is that u cannot narrow down a collection with the ID of a different entity.
- `/api/v1/orders/{orderId}` would be OK
- both options of nick above are correct for filtering by tx, i also slightly prefer the first one, just cause it's an optional field of the orders endpoint, so we don't create additional entities for little gain.

Reply:
I'm in favour of the later option `api/v1/transactions/{txHash}/orders` and `api/v1/transactions/{txHash}/trades`

IMO it's more clear and doesn't mix up with the existing /orders/ endpoint and follows the same pattern as the new user endpoints (`api/v1/account/{owner}/orders|trades`)

I don't expect to have that many results as to need pagination or server side filtering, so I see little benefit in being able to 
> query all orders that sold token X in transaction Y